.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,35 @@
+name: build and test senaite.core
+on:
+  - push
```

Comment:
I think we can remove checks on `push`. It should be enough when this is run on `pull_request`

Reply:
@ramonski that depends on how long pull requests live, for example:

- PR 1: adds new file that uses existing file `A`
- PR 1: runs tests, they pass
- PR 2: deletes file `A`
- PR 2: nobody uses `A` on that branch, tests pass
- PR 2: is merged
- PR 1: is merged, tests fail because `A` is missing

without running tests on `push` you will only notice these failures in the next PR 3.

of course, deleting a file is quite drastic - there are a few more examples that may be more subtle, such as renaming a function.

Reply:
apart from that it allows contributors like me to check if tests pass on their fork **before** they create a pull request :slightly_smiling_face: 

https://github.com/winniehell-forks/senaite.core/actions/workflows/build-and-test.yml

Reply:
Usually we do not merge PRs before all tests have passed and, as far as I know, you can not merge consecutive PRs without having the current HEAD merged before, see here:

https://github.com/senaite/senaite.core/pull/2120/commits/2d22f2182d38175c0db8c3ca142f97b3caa459b7

Regarding test runs, we think indeed that it is good practice to ensure the test pass before a PR is created, but we think that it should be rather done on the local machine instead of outsourcing it to GitHub/TravisCI.

Besides the quota limitations we think it is nicer and a better practice.

Reply:
> Usually we do not merge PRs before all tests have passed

My point was that tests passing on the PR is not sufficient to guarantee they pass on the main branch after merging.
I have build a small example here:

- passing PRs (prior to merge): https://github.com/winniehell/merge-order-test-failure/pulls?q=is%3Apr+is%3Aclosed
- failing main branch (after merge): https://github.com/winniehell/merge-order-test-failure/runs/8265054553

>  as far as I know, you can not merge consecutive PRs without having the current HEAD merged before

I think it is possible and has happened on this project multiple times recently.
However it's had to tell because merge commits are squashed.
It **is** required if merge conflicts occur but test failures can happen without merge conflicts.

Nonetheless I agree with you that it is a good idea to merge the main branch especially into long living branches.
But I wouldn't rely on that. :slightly_smiling_face: 

> we think that it should be rather done on the local machine instead of outsourcing it to GitHub/TravisCI.

Running tests in CI rather than locally is asynchronous, that is I can switch to another branch, close my terminal / IDE or even shut down the computer and still get the test results later.
Additionally running them in CI reduces the risk of debugging "works on my machine"-issues (such as different library versions, uncommited local changes or even a different Python version).

Still I also agree that it's a waste of time to not run the feature tests for the feature you are working on locally.
Again I just wouldn't rely on everybody doing that. :smiley: